R/env.R

```diff
@@ -193,6 +204,78 @@ env_tail <- function(env = env_caller()) {
 }
 
 
+#' Coerce to an environment.
+#'
+#' This is a s3 generic. The default method coerces named vectors
+#' (including lists) to an environment. It first checks that \code{x}
+#' is a dictionary (see \code{\link{is_dict}()}). The method for
+#' strings returns the corresponding package environment (see
+#' \code{\link{env_package}()}).
+#'
+#' @param x An object to coerce.
+#' @param parent A parent environment, \code{\link{env_empty}()} by
+#'   default. Can be ignored with a warning for methods where it does
+#'   not make sense to change the parent.
+#' @export
+#' @examples
+#' # Coerce a named vector to an environment:
+#' env <- as_env(mtcars)
+#'
+#' # By default it gets the empty environment as parent:
+#' identical(env_parent(env), env_empty())
+#'
+#'
+#' # With strings it is a handy shortcut for env_package():
+#' as_env("base")
+#' as_env("rlang")
+#'
+#' # With NULL it returns the empty environment:
+#' as_env(NULL)
+as_env <- function(x, parent = NULL) {
+  UseMethod("as_env")
+}
+
+#' @rdname as_env
+#' @export
+as_env.NULL <- function(x, parent = NULL) {
+  if (!is_null(parent)) {
+    warning("`parent` ignored for empty environment", call. = FALSE)
+  }
+  env_empty()
+}
+
+#' @rdname as_env
+#' @export
+as_env.environment <- function(x, parent = NULL) {
+  if (!is_null(parent)) {
+    env_set_parent(x, parent)
+  }
+  x
+}
+
+#' @rdname as_env
+#' @export
+as_env.character <- function(x, parent = NULL) {
+  if (!is_scalar_character(x)) {
```

Comment:
I think this would be slightly clearer as 

```R
if (length(x) > 1) {
  return(as_env.default(...))
}
```

But it's not clear that length is the right check. What should `as_env(c(a = "b")` return?